core/templates/services/svg-sanitizer.service.ts

```diff
@@ -71,21 +89,33 @@ export class SvgSanitizerService {
       // we need to replace these decimals with a letter so that it's easier
       // to process and validate the filenames.
       if (node.tagName.toLowerCase() === 'svg') {
-        dimensions.height = (
-          (node.getAttribute('height').match(/\d+\.*\d*/g)[0]).replace(
-            '.', 'd'));
-        dimensions.width = (
-          (node.getAttribute('width').match(/\d+\.*\d*/g)[0]).replace(
-            '.', 'd'));
+        let attrHeight = node.getAttribute('height');
+        if (attrHeight) {
+          let heightRegexpMatchResult = attrHeight.match(/\d+\.*\d*/g);
+          if (heightRegexpMatchResult) {
```

Comment:
Is the correct thing to do when either of these checks fail to not set the height? One nice thing about rearranging the code here is it exposes potential failure cases for the sanitizer.

@seanlip any thoughts on if we should be handling null cases in some particular way here? E.g. maybe logging an error or something so that we can be made aware of when these situations occur (vs. just resulting in an undetected broken user experience).

Reply:
I think this depends on the `svgString` we're receiving here. If svgString received does not contain any of the height or width property but it is still a valid svg, then what should be the appropriate behaviour ? Should it throw an error or should it continue execution if any one of them or none exists ? I think svg with no-attributes should scale itself and empty height and width should be assigned in that case. wdyt ?

Reply:
Let's preserve the previous behaviour. I don't think there's a sensible default value, so if the previous behaviour would have resulted in an empty string let's keep that here too.

@mridul-netizen could you try uploading/sanitizing/running an SVG without those properties through this code in the dev server and just seeing what happens? Let us know what you find.

Reply:
@seanlip @BenHenning The particular function is used at math expression input, it is not used for uploading purposes, height and width are assigned on it's own depending on the length of expression. For eg. for an expression like this
`\frac{x}{y}` it creates an svg with default values of height and width ` width="1.784ex" height="3.198ex"`
![image](https://user-images.githubusercontent.com/60519303/126440592-4e073279-a30a-4bcc-9bb2-40a5177adcef.png)

Reply:
This is a general svg sanitizer service. So it needs to support a generic use case. So my question still stands -- try and pass an SVG without the height/width data through this code and see what actually happens. This will tell us whether we should throw an error or not in such cases.

Reply:
Used the function at thumbnail-uploader,
If dimensions are empty:
![Screenshot from 2021-07-21 12-48-51](https://user-images.githubusercontent.com/60519303/126450531-41b000dc-06e0-46bc-a762-ef3a0f819519.png)
If svg contains dimensions:
![Screenshot from 2021-07-21 12-47-32](https://user-images.githubusercontent.com/60519303/126450573-2e0ae79c-7d8b-4d9b-a66d-5c2029c34013.png)

Reply:
@BenHenning As discussed with @seanlip we're now throwing error if height or width attributes are absent.PTAL

Reply:
This seems sensible--thanks!